src/Linear/Primers.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from "react";
+
+import { InputRefFunc } from "../SelectionHandler";
+import { COLOR_BORDER_MAP, darkerColor } from "../colors";
+import { NameRange } from "../elements";
+import { annotation, annotationLabel } from "../style";
+import { FindXAndWidthElementType } from "./SeqBlock";
+
+const hoverOtherPrimerRows = (className: string, opacity: number) => {
+  if (!document) return;
+  const elements = document.getElementsByClassName(className) as HTMLCollectionOf<HTMLElement>;
+  for (let i = 0; i < elements.length; i += 1) {
+    elements[i].style.fillOpacity = `${opacity}`;
+  }
+};
+
+/**
+ * Render each row of annotations into its own row.
+ * This is not a default export for sake of the React component displayName.
+ */
+const PrimeRows = (props: {
+  bpsPerBlock: number;
+  direction: 1 | -1;
+  elementHeight: number;
+  findXAndWidth: FindXAndWidthElementType;
+  firstBase: number;
+  fullSeq: string;
+  inputRef: InputRefFunc;
+  lastBase: number;
+  primerRows: NameRange[][];
+  seqBlockRef: unknown;
+  width: number;
+  yDiff: number;
+}) => (
+  <g>
+    {props.primerRows.map((primers: NameRange[], i: number) => (
+      <PrimerRow
+        key={`annotation-linear-row-${primers[0].id}-${props.firstBase}-${props.lastBase}`}
+        bpsPerBlock={props.bpsPerBlock}
+        direction={props.direction}
+        findXAndWidth={props.findXAndWidth}
+        firstBase={props.firstBase}
+        fullSeq={props.fullSeq}
+        height={props.elementHeight}
+        inputRef={props.inputRef}
+        lastBase={props.lastBase}
+        primers={primers}
+        seqBlockRef={props.seqBlockRef}
+        width={props.width}
+        y={props.yDiff + props.elementHeight * i}
+      />
+    ))}
+  </g>
+);
+
+export default PrimeRows;
+
+/**
+ * A single row of annotations. Multiple of these may be in one seqBlock
+ * vertically stacked on top of one another in non-overlapping arrays.
+ */
+const PrimerRow = (props: {
+  bpsPerBlock: number;
+  direction: 1 | -1;
+  findXAndWidth: FindXAndWidthElementType;
+  firstBase: number;
+  fullSeq: string;
+  height: number;
+  inputRef: InputRefFunc;
+  lastBase: number;
+  primers: NameRange[];
+  seqBlockRef: unknown;
+  width: number;
+  y: number;
+}) => {
+  return (
+    <g
+      className="la-vz-linear-annotation-row"
+      height={props.height * 0.8}
+      transform={`translate(0, ${props.y})`}
+      width={props.width}
+    >
+      {props.primers
+        .filter(a => a.direction == props.direction)
+        .map((a, i) => (
+          <SingleNamedElement
+            {...props} // include overflowLeft in the key to avoid two split annotations in the same row from sharing a key
+            key={`annotation-linear-${a.id}-${i}-${props.firstBase}-${props.lastBase}`}
```

Comment:
```suggestion
            {...props} // include overflowLeft in the key to avoid two split primers in the same row from sharing a key
            key={`primer-linear-${a.id}-${i}-${props.firstBase}-${props.lastBase}`}
```